sdv/utils.py

```diff
@@ -15,6 +15,17 @@ def cast_to_iterable(value):
     return [value]
 
 
+def get_fisrt_valid_value(values):
+    """Return the first not ``nan`` value or ``nan`` if all are ``nan``."""
+    for item in values:
+        if pd.isna(item):
+            continue
+        elif item != 'nan':
+            return item
```

Comment:
minor but you can rewrite as:
```python
for item in values:
    if !pd.isna(item):
        return item

return item

Reply:
The problem comes with parsed strings, I think I even need to increase the below condition with 'nan', 'NaT', 'None' etc.
The pd.isna does not consider those a nan value as they are string representation of them. But I can simplify it a bit and write it in one if.